runtime/src/main/java/org/corfudb/runtime/Backup.java

```diff
@@ -265,11 +282,36 @@ private void addToTarFile(File tableFile,TarArchiveOutputStream tarOutput) throw
                 .map(tableName -> CorfuRuntime
                         .getStreamID(TableRegistry.getFullyQualifiedTableName(tableName)))
                 .collect(Collectors.toList());
-        log.info("{} tables need to be backed up: {}", tables.size(), tables);
+        log.info("{} tables need to be backed up.", tables.size());
 
         return tables;
     }
 
+    /**
+     * Get UUIDs of all registered UFO tables
+     *
+     * @return  List<UUID>   - a list of UUIDs for all registered UFO tables
+     */
+    private List<UUID> getAllTablesUUID() {
+        return runtime.getTableRegistry().listTables().stream()
+                .map(table -> CorfuRuntime.getStreamID(
+                        TableRegistry.getFullyQualifiedTableName(table.getNamespace(), table.getTableName())))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Get a map which maps stream ID to TableName
+     *
+     * @return  Map<>UUID, String</>   - a map: streamId -> fully qualified table name
+     */
+    private Map<UUID, String> getStreamIdToTableNameMap() {
+        Map<UUID, String> streamIdToTableNameMap = new HashMap<>();
+        runtime.getTableRegistry().listTables().forEach(tableName -> {
+                String name = TableRegistry.getFullyQualifiedTableName(tableName);
+                streamIdToTableNameMap.put(CorfuRuntime.getStreamID(name), name);
+        });
+        return streamIdToTableNameMap;
+    }
```

Comment:
I think `putIfAbsent ()` inside forEach needs to be synchronized, so introducing parallelism is not that helpful right?

Reply:
@xcchang suggested to use a ConcurrentHashMap. It can be parallelized here, but I think it won't benefit too much from the parallelism - the task it not cpu intensive.